crates/bevy_core_pipeline/src/bloom/mod.rs

```diff
@@ -637,8 +641,8 @@ fn prepare_bloom_textures(
         }) = camera.physical_viewport_size
         {
             let min_view = width.min(height) as f32;
-            // How many times we can halve the resolution, minus 3 to avoid tiny mips
-            let mip_count = (min_view.log2().round() as i32 - 3).max(1) as u32;
+            // How many times we can halve the resolution
+            let mip_count = (min_view.log2().floor() as i32).max(1) as u32;
```

Comment:
Why did you change this to floor, and remove the -3? 

If we keep this, it should be ` let mip_count = (min_view.log2().floor() as u32).max(1);`

Reply:
Round sometimes results in too many mips (one of the dimensions below 1) which causes an error on the gpu.

Removed -3 so that we get closer to the smallest mip being 1x1. We need it if our definition of bloom intensity is how likely the light is to exit at a random point of the lens. Intensity of 1 should result in the entire screen being the same color (average of the original image). The change, however, still does not result in 1x1 textures though because when one of the dimensions hits 1 we cant go further. This is not an ideal solution but it is still closer to the irl counterpart

Reply:
Ok, sounds good to me. Then you just need to fix the u32 casting like I mentioned above.

Reply:
We will also need to test this on a more complex example. The bloom example in bevy originally looked fine, but testing on other scenes revealed that we needed to do -3 to avoid artifacts. Also, going down to 1x1 may not be worth it. It's probably fine to settle on 4x4 or something, and be mostly accurate, while avoiding the extra draw calls.